.circleci/config.yml

```diff
@@ -0,0 +1,112 @@
+orbs:
+  win: circleci/windows@1.0.0
+
+version: 2.1
+
+executors:
+  golang:
+    docker:
+      - image: docker.mirror.hashicorp.services/circleci/golang:1.16
+    resource_class: medium+
+  darwin:
+    macos:
+      xcode: "12.0.0"
+
+commands:
+  install-go-run-tests-unix:
+    parameters:
+      GOOS:
+        type: string
+      GOVERSION:
+        type: string
+      HOME:
+        type: string
+        default: "~"
+    steps:
+      - checkout
+      - run: curl https://dl.google.com/go/go<< parameters.GOVERSION >>.<< parameters.GOOS >>-amd64.tar.gz | tar -C << parameters.HOME >>/ -xz
+      - run: << parameters.HOME >>/go/bin/go test ./... -coverprofile=coverage.txt  -covermode=atomic
+  install-go-run-tests-windows:
+    parameters:
+      GOVERSION:
+        type: string
+    steps:
+      - checkout
+      - run: curl https://dl.google.com/go/go<< parameters.GOVERSION >>.windows-amd64.zip --output ~/go<< parameters.GOVERSION >>.windows-amd64.zip
+      - run: unzip ~/go<< parameters.GOVERSION >>.windows-amd64.zip -d ~/
+      - run: ~/go/bin/go test ./... -coverprofile=coverage.txt  -covermode=atomic
+  build-and-persist-plugin-binary:
+    parameters:
+      GOOS:
+        type: string
+      GOARCH:
+        default: "amd64"
+        type: string
+    steps:
+      - checkout
+      - run: GOOS=<< parameters.GOOS >> GOARCH=<<parameters.GOARCH>> go build -o ./pkg/packer_plugin_amazon_<< parameters.GOOS >>_<< parameters.GOARCH >> .
+      - run: zip ./pkg/packer_plugin_amazon_<< parameters.GOOS >>_<< parameters.GOARCH >>.zip ./pkg/packer_plugin_amazon_<< parameters.GOOS >>_<< parameters.GOARCH >>
+      - run: rm ./pkg/packer_plugin_amazon_<< parameters.GOOS >>_<< parameters.GOARCH >>
+      - persist_to_workspace:
+          root: .
+          paths:
+            - ./pkg/
+
+# Golang CircleCI 2.0 configuration file
+#
+# Check https://circleci.com/docs/2.0/language-go/ for more details
+jobs:
+  test-linux:
+    executor: golang
+    resource_class: large
+    working_directory: /go/src/github.com/hashicorp/packer-plugin-amazon
+    steps:
+      - checkout
+      - run: TESTARGS="-coverprofile=coverage.txt  -covermode=atomic" make ci
+  test-darwin:
+    executor: darwin
+    working_directory: ~/go/github.com/hashicorp/packer-plugin-amazon
+    steps:
+      - install-go-run-tests-unix:
+          GOOS: darwin
+          GOVERSION: "1.16"
+  test-windows:
+    executor:
+      name: win/vs2019
+      shell: bash.exe
+    steps:
+      - install-go-run-tests-windows:
+          GOVERSION: "1.16"
+  check-lint:
+    executor: golang
+    resource_class: xlarge
```

Comment:
We can probably drop this down now that it is just the plugin code. But that is just me thinking out loud.

Reply:
I think you're right!